providers/implementations/kem/build.info

```diff
@@ -20,7 +20,7 @@ IF[{- !$disabled{'ml-kem'} -}]
   IF[{- !$disabled{ec} -}]
     SOURCE[$TLS_ML_KEM_HYBRID_GOAL]=mlx_kem.c
   ENDIF
-  SOURCE[$ML_KEM_GOAL] = ml_kem.c
+  SOURCE[$ML_KEM_GOAL] = ml-kem_kem.c
```

Comment:
Please DO NOT rename the file, this creates pointless conflicts.

Reply:
Keeping the file name caused a build error because there are then two files named `ml_kem.c` in the FIPS provider.  I had to rename one of the two…

Reply:
Sounds like a problem with the FIPS module build code, and not the source file name. I see no source file conflicts. Are these the "conflicting" files?
```
crypto/ml_kem/ml_kem.c
providers/implementations/kem/ml_kem.c
```
If the FIPS module build expects files from different source directories to have distinct relative names, could we fix the faulty build logic instead.

Since we're using `ml_kem...`, `ml_dsa..` in all the other file names, it is odd to suddenly use `ml-kem_kem.c`, how about `kem-ml_kem.c`, but only after encoders and decoders PR goes in.

Reply:
The build system expects unique files names.  I'm surprised the problem didn't sting you earlier.

I'm okay with that file name.  The other I thought of was `ml_kem_kem.c` which matches the other files `algorithm_kem.c`

Reply:
The `ml_kem_kem.c` version is also OK, be it less clear why we're repeating ourselves. Of course the real problem is still the "unique" file name requirement, where is that coming from. There's no such problem outside the FIPS module, and indeed there has been no issue till now.

Reply:
> I'm okay with that file name. The other I thought of was `ml_kem_kem.c` which matches the other files `algorithm_kem.c`

That would be my preference - to align the name with other algorithms. The `kem` duplication is IMO obvious once you look at the other files in the subdirectory.

Reply:
That's what I did.